source/test/socket_abstract_test.cpp

```diff
@@ -419,8 +415,23 @@ int socket_api_test_echo_client_connected(socket_stack_t stack, socket_address_f
         }
         // Override event for dgrams.
         if (pf == SOCKET_DGRAM) {
+
+        	err = api->bind(&s, &local_ipaddr, local_port);
+            if (!TEST_EQ(err, SOCKET_ERROR_NONE)) {
+                TEST_RETURN();
+            }
+
             client_event.event = SOCKET_EVENT_CONNECT;
             client_event_done = true;
+
+            api->get_local_addr(&s, &local_ipaddr);
+            api->get_local_port(&s, &local_port);
+            TEST_PRINT(">>> MBED TARGET ipaddr:port=%x.%x.%x.%x:%d\r\n",
```

Comment:
This ```printf()``` will probably not have the expected results. ```ipv6be``` is a big-endian IPv6 address. This means that the IPv4 address is entirely contained within ```ipv6be[3]```.  It might be useful to use ```inet_ntop``` here.